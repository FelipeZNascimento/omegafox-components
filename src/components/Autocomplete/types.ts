export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
