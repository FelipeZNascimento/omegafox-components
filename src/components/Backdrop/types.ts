export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
