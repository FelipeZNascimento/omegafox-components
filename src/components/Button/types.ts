export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
