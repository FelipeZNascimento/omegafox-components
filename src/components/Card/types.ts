export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
