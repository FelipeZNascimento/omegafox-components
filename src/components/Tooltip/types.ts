export interface ITooltipProps {
  children: JSX.Element;
  position: 'top' | 'right' | 'bottom' | 'left';
  text: string;
}
