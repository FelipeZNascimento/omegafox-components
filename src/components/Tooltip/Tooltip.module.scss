@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/global.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.tooltip {
  cursor: pointer;
  display: inline-block;
  position: relative;

  &:hover .tooltipText {
    opacity: 1;
    visibility: visible;
  }
}

.tooltipText {
  @include aligned(center);
  @include bg-rgba-colored($color-black, 0.6);
  @include font-colored($color-white);
  @include font-sized(s);
  @include padded(all, s2);

  visibility: hidden;
  opacity: 0;
  position: absolute;
  transition: opacity $transition-time;
  width: 120px;
  z-index: 1;

  &Top {
    bottom: 105%;
    border-radius: $border-radius;
    border-bottom-left-radius: 0;
    left: 50%;
  }

  &Right {
    border-radius: $border-radius;
    border-bottom-left-radius: 0;
    left: 105%;
    top: -40%;
  }

  &Bottom {
    border-radius: $border-radius;
    border-top-left-radius: 0;
    left: 50%;
    top: 105%;
  }

  &Left {
    border-radius: $border-radius;
    border-bottom-right-radius: 0;
    top: -40%;
    right: 105%;
  }
}
