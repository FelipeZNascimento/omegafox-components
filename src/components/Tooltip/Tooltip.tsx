import React from 'react';
import { isMobile } from 'react-device-detect';

import classNames from 'classnames';
import { ITooltipProps } from './types';
import styles from './Tooltip.module.scss';

export const Tooltip = ({ children, position, text }: ITooltipProps) => {
  if (isMobile) {
    return children;
  }
  const tooltipTextClass = classNames(styles.tooltipText, {
    [styles.tooltipTextTop]: position === 'top',
    [styles.tooltipTextRight]: position === 'right',
    [styles.tooltipTextBottom]: position === 'bottom',
    [styles.tooltipTextLeft]: position === 'left'
  });

  return (
    <div className={styles.tooltip}>
      {children}
      <span className={tooltipTextClass}>{text}</span>
    </div>
  );
};
