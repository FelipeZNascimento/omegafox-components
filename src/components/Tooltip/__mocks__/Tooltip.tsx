import React from 'react';
import { isMobile } from 'react-device-detect';

import { ITooltipProps } from '../types';

export const Tooltip = ({
  children,
  position = 'bottom',
  text,
  type = 'dark'
}: ITooltipProps) => {
  if (isMobile) {
    return children;
  }

  return (
    <div>
      <span data-tip={text}>{children}</span>
      <p>position: {position}</p>
      <p>type: {type}</p>
    </div>
  );
};
