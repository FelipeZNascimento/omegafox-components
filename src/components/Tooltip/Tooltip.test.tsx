import React from 'react';
import { render } from '@testing-library/react';
import { Tooltip } from './Tooltip';
import { ITooltipProps } from './types';

describe('<Tooltip />', () => {
  const renderComponent = ({ position, text }: ITooltipProps) =>
    render(<Tooltip position={position} text={text} />);

  it('should render', () => {
    const { container } = renderComponent({
      position: 'top',
      text: 'Tooltip text'
    });

    expect(container).toMatchSnapshot();
  });
});
