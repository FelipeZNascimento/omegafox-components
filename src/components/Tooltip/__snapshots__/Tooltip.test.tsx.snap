// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render 1`] = `
<div>
  <div
    class="tooltip"
  >
    <span>
      Hover to see tooltip
    </span>
    <span
      class="tooltipText tooltipTextTop"
    >
      Tooltip text
    </span>
  </div>
</div>
`;
